packages/frontend/src/api.ts

```diff
@@ -38,13 +36,17 @@ type LightdashApiProps = {
     url: string;
     body: BodyInit | null | undefined;
     headers?: Record<string, string> | undefined;
+    version?: 'v1' | 'v2';
 };
 export const lightdashApi = async <T extends ApiResponse['results']>({
     method,
     url,
     body,
     headers,
+    version = 'v1',
 }: LightdashApiProps): Promise<T> => {
+    const apiPrefix = `${BASE_API_URL}api/${version}`;
```

Comment:
I like this a lot more than I have on my PR: https://github.com/lightdash/lightdash/pull/10758/files